tree/dataframe/inc/ROOT/RActionSnapshot.hxx

```diff
@@ -0,0 +1,257 @@
+#ifndef ROOT_RACTIONSNAPSHOT
```

Comment:
I think this is supposed to showcase the brillant idea, but I'd drop this nevertheless: this code would need to be streamlined in the already existing structure and enriched perhaps with tests (which could well be in first approx the same tests of snapshot)

Reply:
tests are there: all the tests for jitted snapshots now test this code.

i would not add hundreds of lines to the already large `RDFNodes.hxx`, but rather continue with the effort of dividing it in several smaller files. These specializations of `RAction` would end up in the same file as the main `RAction` definition.

but more than anything, I would love to not have to repeat all of `RAction`'s definition three times :sweat_smile: but I don't know how to do it, I have to think about it. hard.